File: DemoMethods/src/main/java/net/ravendb/demo/indexes/LastFmAnalyzed.java
package net.ravendb.demo.indexes;

import com.mysema.query.annotations.QueryEntity;
import net.ravendb.abstractions.indexing.FieldIndexing;
import net.ravendb.client.indexes.AbstractIndexCreationTask;

public class LastFmAnalyzed extends AbstractIndexCreationTask {

    @QueryEntity
    public static class Result {
        private String query;

        public String getQuery() {
            return query;
        }

        public void setQuery(String query) {
            this.query = query;
        }
    }

    public LastFmAnalyzed() {

        QLastFmAnalyzed_Result result = QLastFmAnalyzed_Result.result;

        map = "from song in docs.LastFms " +
                "select new { " +
                "  Query = new object[] { " +
                "    song.Artist, " +
                "    ((object)song.TimeStamp), " +
                "    song.Tags, " +
                "    song.Title, " +
                "    song.TrackId " +
                "  } " +
                "}";

        index(result.query, FieldIndexing.ANALYZED);
    }
}

File: DemoServer/Indexes/LastFmAnalyzed.cs
using System.Linq;
using DemoServer.Entities;
using Raven.Abstractions.Indexing;
using Raven.Client.Indexes;

namespace DemoServer.Indexes
{
    public class LastFmAnalyzed : AbstractIndexCreationTask<LastFm, LastFmAnalyzed.Result>
    {
        public class Result
        {
            public string Query;
        }

        public LastFmAnalyzed()
        {
            Map = songs => from song in songs
                select new
                {
                    Query = new object[]
                    {
                        song.Artist, 
                        song.TimeStamp, 
                        song.Tags, 
                        song.Title, 
                        song.TrackId
                    }
                };

            Index(x => x.Query, FieldIndexing.Analyzed);
        }
    }
}